### PR TITLE
Unified API Store view for APIs and API Products

### PR DESCRIPTION
**Pull request by uvindra**
Unified_Store

APIs and API Products will share the same view and functional windows on the API Store

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/model/ApiTypeWrapper.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
+package org.wso2.carbon.apimgt.api.model;
```
Comment (ruks) [inline components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/model/ApiTypeWrapper.java]: missing license header

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/AbstractAPIManager.java [hunk 2]
```diff
@@ -516,6 +518,61 @@ public API getAPIbyUUID(String uuid, String requestedTenantDomain) throws APIMan
...
+                    return new ApiTypeWrapper(getApiForPublishing(registry, apiArtifact));
+                }
+            } else {
+                String msg = "Failed to get API. API artifact corresponding to artifactId " + uuid + " does not exist";
+                log.error(msg);
+                throw new APIMgtResourceNotFoundException(msg);
+            }
+        } catch (RegistryException e) {
+            String msg = "Failed to get API";
+            log.error(msg, e);
```
Comment (ruks) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/AbstractAPIManager.java]: I think we no need to log the error here.

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/executors/APIExecutor.java [hunk 3]
```diff
@@ -229,5 +143,110 @@ public boolean execute(RequestContext context, String currentState, String targe
...
+            throws APIManagementException, FaultGatewaysException, RegistryException {
+        boolean executed;
+
+        String oldStatus = APIUtil.getLcStateFromArtifact(apiArtifact);
+        String newStatus = (targetState != null)? targetState.toUpperCase(): targetState;
+
+        boolean isCurrentCreatedOrPrototyped = APIConstants.CREATED.equals(oldStatus) ||
+                APIConstants.PROTOTYPED.equals(oldStatus);
+        boolean isStateTransitionToPublished = isCurrentCreatedOrPrototyped && APIConstants.PUBLISHED.equals(newStatus);
+        if(newStatus != null){ //only allow the executor to be used with default LC states transition
```
Comment (ruks) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/executors/APIExecutor.java]: seems formatting issue

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIAPIProductNameComparator.java [hunk 4]
```diff
@@ -0,0 +1,103 @@
+package org.wso2.carbon.apimgt.impl.utils;
```
Comment (ruks) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIAPIProductNameComparator.java]: missing license header

File: components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/model/ApiTypeWrapper.java (hunk 1 above)
Comment (uvindra) [reply components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/model/ApiTypeWrapper.java]: Fixed

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/AbstractAPIManager.java (hunk 2 above)
Comment (uvindra) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/AbstractAPIManager.java]: Fixed

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/executors/APIExecutor.java (hunk 3 above)
Comment (uvindra) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/executors/APIExecutor.java]: Fixed

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIAPIProductNameComparator.java (hunk 4 above)
Comment (uvindra) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIAPIProductNameComparator.java]: Fixed

**Pull request merged**